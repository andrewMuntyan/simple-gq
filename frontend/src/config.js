export const endpoint = `http://localhost:4000`;
export const prodEndpoint = ``;
export const WORDS_PER_PAGE = 3;
