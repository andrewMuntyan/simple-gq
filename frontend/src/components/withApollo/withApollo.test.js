import React from 'react';
import { shallow } from 'enzyme';
import { ApolloClient } from 'apollo-client';
import { HttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import toJSON from 'enzyme-to-json';

import { endpoint } from '../../config';

import { withApollo } from '.';

jest.mock('apollo-link-http');
jest.mock('apollo-cache-inmemory');

const BaseComponent = () => <p>base</p>;
const WrappedComponent = withApollo(BaseComponent);

describe('<WithApollo />', () => {
  const wrapper = shallow(<WrappedComponent />);

  it('renders and displays properly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should wrap base component with ApolloProvider', () => {
    const ApolloProvider = wrapper.find('ApolloProvider');
    expect(ApolloProvider).toHaveLength(1);
  });

  it('ApolloProvider should be cobfigured with ApolloClient instance', () => {
    const ApolloProvider = wrapper.find('ApolloProvider');
    const {
      client,
      client: { link, cache }
    } = ApolloProvider.props();
    expect(client).toBeInstanceOf(ApolloClient);
    expect(link).toBeInstanceOf(HttpLink);
    expect(cache).toBeInstanceOf(InMemoryCache);
    expect(HttpLink).toHaveBeenCalledTimes(1);
    expect(HttpLink).toHaveBeenCalledWith({
      uri: endpoint
    });
  });
});
