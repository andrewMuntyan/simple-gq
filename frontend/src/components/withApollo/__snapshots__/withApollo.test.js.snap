// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithApollo /> renders and displays properly 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "broadcastWatches": [MockFunction],
        "diff": [MockFunction],
        "evict": [MockFunction],
        "extract": [MockFunction],
        "maybeBroadcastWatch": [MockFunction],
        "performTransaction": [MockFunction],
        "read": [MockFunction],
        "readFragment": [MockFunction],
        "readQuery": [MockFunction],
        "recordOptimisticTransaction": [MockFunction],
        "removeOptimistic": [MockFunction],
        "reset": [MockFunction],
        "restore": [MockFunction],
        "transformDocument": [MockFunction],
        "transformForLink": [MockFunction],
        "watch": [MockFunction],
        "write": [MockFunction],
        "writeData": [MockFunction],
        "writeFragment": [MockFunction],
        "writeQuery": [MockFunction],
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": HttpLink {
        "concat": [MockFunction],
        "request": [MockFunction],
        "split": [MockFunction],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "broadcastWatches": [MockFunction],
          "diff": [MockFunction],
          "evict": [MockFunction],
          "extract": [MockFunction],
          "maybeBroadcastWatch": [MockFunction],
          "performTransaction": [MockFunction],
          "read": [MockFunction],
          "readFragment": [MockFunction],
          "readQuery": [MockFunction],
          "recordOptimisticTransaction": [MockFunction],
          "removeOptimistic": [MockFunction],
          "reset": [MockFunction],
          "restore": [MockFunction],
          "transformDocument": [MockFunction],
          "transformForLink": [MockFunction],
          "watch": [MockFunction],
          "write": [MockFunction],
          "writeData": [MockFunction],
          "writeFragment": [MockFunction],
          "writeQuery": [MockFunction],
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "broadcastWatches": [MockFunction],
            "diff": [MockFunction],
            "evict": [MockFunction],
            "extract": [MockFunction],
            "maybeBroadcastWatch": [MockFunction],
            "performTransaction": [MockFunction],
            "read": [MockFunction],
            "readFragment": [MockFunction],
            "readQuery": [MockFunction],
            "recordOptimisticTransaction": [MockFunction],
            "removeOptimistic": [MockFunction],
            "reset": [MockFunction],
            "restore": [MockFunction],
            "transformDocument": [MockFunction],
            "transformForLink": [MockFunction],
            "watch": [MockFunction],
            "write": [MockFunction],
            "writeData": [MockFunction],
            "writeFragment": [MockFunction],
            "writeQuery": [MockFunction],
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": HttpLink {
          "concat": [MockFunction],
          "request": [MockFunction],
          "split": [MockFunction],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "broadcastWatches": [MockFunction],
            "diff": [MockFunction],
            "evict": [MockFunction],
            "extract": [MockFunction],
            "maybeBroadcastWatch": [MockFunction],
            "performTransaction": [MockFunction],
            "read": [MockFunction],
            "readFragment": [MockFunction],
            "readQuery": [MockFunction],
            "recordOptimisticTransaction": [MockFunction],
            "removeOptimistic": [MockFunction],
            "reset": [MockFunction],
            "restore": [MockFunction],
            "transformDocument": [MockFunction],
            "transformForLink": [MockFunction],
            "watch": [MockFunction],
            "write": [MockFunction],
            "writeData": [MockFunction],
            "writeFragment": [MockFunction],
            "writeQuery": [MockFunction],
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "broadcastWatches": [MockFunction],
          "diff": [MockFunction],
          "evict": [MockFunction],
          "extract": [MockFunction],
          "maybeBroadcastWatch": [MockFunction],
          "performTransaction": [MockFunction],
          "read": [MockFunction],
          "readFragment": [MockFunction],
          "readQuery": [MockFunction],
          "recordOptimisticTransaction": [MockFunction],
          "removeOptimistic": [MockFunction],
          "reset": [MockFunction],
          "restore": [MockFunction],
          "transformDocument": [MockFunction],
          "transformForLink": [MockFunction],
          "watch": [MockFunction],
          "write": [MockFunction],
          "writeData": [MockFunction],
          "writeFragment": [MockFunction],
          "writeQuery": [MockFunction],
        },
      },
      "typeDefs": undefined,
      "version": "2.6.4",
      "watchQuery": [Function],
    }
  }
>
  <BaseComponent />
</ApolloProvider>
`;
